File: app/src/main/java/com/mrebollob/loteriadenavidad/app/LotteryApplication.java
package com.mrebollob.loteriadenavidad.app;

import android.app.Application;

import com.mrebollob.loteriadenavidad.app.di.components.AppComponent;
import com.mrebollob.loteriadenavidad.app.di.components.DaggerAppComponent;
import com.mrebollob.loteriadenavidad.app.di.modules.AppModule;

/**
 * @author mrebollob
 */
public class LotteryApplication extends Application {

    private AppComponent mAppComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        initializeInjector();
    }

    private void initializeInjector() {
        mAppComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
    }

    public AppComponent getAppComponent() {
        return mAppComponent;
    }
}

File: androidApp/src/main/java/com/mrebollob/loteria/android/presentation/platform/LotteryApplication.kt
package com.mrebollob.loteria.android.presentation.platform

import android.app.Application
import co.touchlab.kermit.Logger
import com.mrebollob.loteria.android.BuildConfig
import com.mrebollob.loteria.android.analytics.AnalyticsManager
import com.mrebollob.loteria.android.di.appModule
import com.mrebollob.loteria.di.initKoin
import org.koin.android.ext.android.inject
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.logger.Level

class LotteryApplication : Application() {

    private val analyticsManager: AnalyticsManager by inject()

    override fun onCreate() {
        super.onCreate()
        initKoin {
            // https://github.com/InsertKoinIO/koin/issues/1188
            androidLogger(if (BuildConfig.DEBUG) Level.ERROR else Level.NONE)
            androidContext(this@LotteryApplication)
            modules(appModule)
        }

        initAnalytics()
        Logger.d { "LotteryApplication" }
    }

    private fun initAnalytics() {
        analyticsManager.setUserData()
    }
}
